pest/src/error.rs

```diff
@@ -457,7 +459,8 @@ mod tests {
                 "  |  ^---",
                 "  |",
                 "  = unexpected 4, 5, or 6; expected 1, 2, or 3",
-            ].join("\n")
+            ]
```

Comment:
Same with these. I guess it would be best to use something better than `join`ing here.

Reply:
The "other" option:

```rust
        assert_eq!(
            format!("{}", error),
            " --> 2:2
  |
2 | cd
  |  ^---
  |
  = unexpected 4, 5, or 6; expected 1, 2, or 3",
        );
```